spot_wrapper/wrapper.py

```diff
@@ -601,90 +1030,132 @@ def _robot_command(self, command_proto, end_time_secs=None, timesync_endpoint=No
             timesync_endpoint: (optional) Time sync endpoint
         """
         try:
-            id = self._robot_command_client.robot_command(lease=None, command=command_proto,
-                                                          end_time_secs=end_time_secs,
-                                                          timesync_endpoint=timesync_endpoint)
-            return True, "Success", id
+            command_id = self._robot_command_client.robot_command(
+                lease=None,
+                command=command_proto,
+                end_time_secs=end_time_secs,
+                timesync_endpoint=timesync_endpoint,
+            )
+            return True, "Success", command_id
         except Exception as e:
-            self._logger.error("Unable to execute robot command, exception was" + str(e))
+            self._logger.error(
+                "Unable to execute robot command, exception was" + str(e)
+            )
             return False, str(e), None
 
+    @try_claim
     def stop(self):
         """Stop the robot's motion."""
-        response = self.claim()
-        if not response[0]:
-            return response
         response = self._robot_command(RobotCommandBuilder.stop_command())
         return response[0], response[1]
 
+    @try_claim(power_on=True)
     def self_right(self):
         """Have the robot self-right itself."""
-        response = self.power_on(allow_already_powered=True)
-        if not response[0]:
-            return response
         response = self._robot_command(RobotCommandBuilder.selfright_command())
         return response[0], response[1]
 
+    @try_claim
```

Comment:
```suggestion
    @try_claim(power_on=True)
```

Reply:
I think I understand why this might be necessary - if the robot is on the floor but not in the correct sitting position (e.g. legs in bad orientation) and not powered on, this would be applicable. I was thinking that if you are trying to sit then you must by definition be standing, but that's not necessarily the case.